plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -624,8 +624,11 @@ void chain_plugin_impl::plugin_initialize(const variables_map& options) {
         resmon_plugin->monitor_directory(chain_config->state_dir);
       }
 
-      if( options.count( "chain-state-db-size-mb" ))
+      if( options.count( "chain-state-db-size-mb" )) {
          chain_config->state_size = options.at( "chain-state-db-size-mb" ).as<uint64_t>() * 1024 * 1024;
+         EOS_ASSERT( chain_config->state_size <= 8ull * 1024 * 1024 * 1024 * 1024, plugin_config_exception,
+                     "The maximum supported size for the chain state db is 8TiB" );
```

Comment:
imo it would be nice for this error message to expressly refer to `chain-state-db-size-mb` somehow, so the user knows exactly what knob is set too far. But, users can probably figure it out in this case anyways